From 9d85bc2294e81803ce915c657b86aebb86aef310 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Fri, 18 Aug 2023 10:50:12 +0200
Subject: [PATCH] On CUDA 12, pass -arch=all to compile for all supported
 architectures

---
 Makefile.am | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 2237a7fb..874349d8 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -15,15 +15,7 @@ AM_NVCC_FLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/src -Xptxas="${PTX_FLAGS
 else
 AM_NVCC_FLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/src -Xptxas="${PTX_FLAGS}"
 endif
-AM_NVCC_FLAGS += -gencode arch=compute_52,code=sm_52
-AM_NVCC_FLAGS += -gencode arch=compute_60,code=sm_60
-AM_NVCC_FLAGS += -gencode arch=compute_61,code=sm_61
-AM_NVCC_FLAGS += -gencode arch=compute_70,code=sm_70
-AM_NVCC_FLAGS += -gencode arch=compute_75,code=sm_75
-AM_NVCC_FLAGS += -gencode arch=compute_80,code=sm_80
-AM_NVCC_FLAGS += -gencode arch=compute_86,code=sm_86
-AM_NVCC_FLAGS += -gencode arch=compute_87,code=sm_87
-AM_NVCC_FLAGS += -gencode arch=compute_86,code=compute_86
+AM_NVCC_FLAGS += -arch=all
 AM_LD_FLAGS = -lcuda
 NVCCLINK = \
 	$(NVCC) $(NVCCFLAGS) $(AM_NVCC_FLAGS) $(AM_LD_FLAGS) $(LDFLAGS) $(OPENMP_LIB) -o $@

